src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -211,6 +220,60 @@ public boolean hasSchedule(Schedule schedule) {
         return addressBook.hasSchedule(schedule);
     }
 
+    /**
+     * Returns true if a Schedule with the same client as {@code Client} exists in the Schedule List.
+     */
+    public boolean hasAnyScheduleAssociatedWithClient(Client clientToEdit) {
+        requireNonNull(clientToEdit);
+        return addressBook.getScheduleList()
+                .stream()
+                .map(Schedule::getClient)
+                .anyMatch(clientToEdit::isUnique);
+    }
+
+    /**
+     * Edits every Schedule with the same client as {@code client}.
+     */
+    public void editSchedulesAssociatedWithClient(Client clientToEdit, Client editedClient) {
+        requireAllNonNull(clientToEdit, editedClient);
+
+        List<Schedule> associatedSchedules = addressBook.getScheduleList()
+                .stream()
+                .filter(schedule -> clientToEdit.isUnique(schedule.getClient()))
+                .collect(Collectors.toList());
+
+        for (Schedule schedule : associatedSchedules) {
+            this.setSchedule(schedule, new Schedule(editedClient, schedule.getSession()));
+        }
+    }
+
+    /**
+     * Returns true if a Schedule with the same session as {@code session} exists in the Schedule List.
+     */
+    public boolean hasAnyScheduleAssociatedWithSession(Session session) {
+        requireNonNull(session);
+        return addressBook.getScheduleList()
+                .stream()
+                .map(Schedule::getSession)
+                .anyMatch(session::isUnique);
+    }
```

Comment:
Repetition of line 151

Reply:
Think this duplication happened from merging from both @dhafinrazaq and your branch? Will keep hasAnyScheduleAssociatedWithSession()

Reply:
ook!